### PR TITLE
feat: include onprem cluster depl modules in new infra infra v2

### PR DESCRIPTION
**Pull request by alouvion-aneo**
the goal of this pr is to add a mudule to deploy a kubernetes cluster on a given list of VMs.
Basically you need in input : ip + ssh key and it will deploy kubeadm / network cni / kubeadm join / ...

### CODE REVIEW

File: kubernetes/onpremise/kubeadm/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<!-- BEGIN_TF_DOCS -->
```
[the remark this answers is not in the document]
Comment (alouvion-aneo) [reply kubernetes/onpremise/kubeadm/README.md:12]: ok

File: kubernetes/onpremise/kubeadm/scripts/install_docker.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# upgrade distrib
```
[the remark this answers is not in the document]
Comment (alouvion-aneo) [reply kubernetes/onpremise/kubeadm/scripts/install_docker.sh:1]: ok

File: kubernetes/onpremise/kubeadm/variables.tf [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  description = "The worker nodes to be deployed."
+  type = map(object({
+    instance_count = optional(number, 1)
+    label          = optional(list(string), [])
+    name           = string
+    public_dns     = string
+    taints         = optional(list(string), [])
+  }))
+}
+#ex:
```
[the remark this answers is not in the document]
Comment (alouvion-aneo) [reply kubernetes/onpremise/kubeadm/variables.tf]: ok

File: kubernetes/onpremise/kubeadm/exemples/complete/main.tf [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  special = false
+  upper   = false
+}
+
+# right part of the kubeadm token.
+resource "random_string" "token_secret" {
+  length  = 16
+  special = false
+  upper   = false
+}
```
Comment (lzianekhodja) [inline kubernetes/onpremise/kubeadm/exemples/complete/main.tf:54]: We cannot execute this example If these VMs do not exist.

**Review by lzianekhodja — APPROVED** [review]

**Pull request merged**